shared/js/background/https.es6.js

```diff
@@ -1,152 +1,72 @@
-        // serve a different list if needed:
-        if (window.safari) {
-            url += '&b=safari'
+    // Sets a list by type and name. This is data that
+    // is gathered from HTTPSStorage.
+    // 'upgrade list' is assumed to be a bloom filter
+    // 'whitelist' is an array
+    setLists (lists) {
+        try {
+            lists.map(list => {
+                if (!list.data) {
+                    throw new Error(`HTTPS: ${list.name} missing data`)
+                }
+
+                if (list.type === 'upgrade list') {
+                    this.upgradeLists.set(list.name, this.createBloomFilter(list))
+                } else if (list.type === 'whitelist') {
+                    this.whitelist = list.data
+                }
+            })
+            this.isReady = true
+            console.log('HTTPS: is ready')
+        } catch (e) {
+            // a failed setLists update will turn https off
+            // validation of the data should happen before calling setLists
+            this.isReady = false
+            console.log('HTTPS: setLists error, not ready')
+            console.log(e)
         }
-
-        console.log('HTTPS: updateList() check if new list exists at: ' + url)
-
-        // try to load an updated file from the server, passing
-        // in the latest etag we have and only calling the callback
-        // with the new file if the etag on the server is different:
-        load.loadExtensionFile({
-            url: url,
-            source: 'external',
-            etag: etag
-        }).then((res) => {
-            // This only gets called if the etag is different
-            // and it was able to get a new list from the server:
-            console.log('HTTPS: updateList() got updated list from server')
-
-            let data = res.data
-            let newEtag = res.getResponseHeader('etag') || ''
-
-            try {
-                // update list in memory
-                let parsedData = JSON.parse(data)
-
-                httpsUpgradeList = parsedData
-                console.log('HTTPS: updateList() new list has ' + httpsUpgradeList.length + ' items')
-
-                // save the full data response to storage
-                // so we don't have to re-stringify the parsed JSON object:
-                this.saveToStorage(data)
-
-                // save new etag for next time
-                settings.updateSetting('https-etag', newEtag)
-                console.log('HTTPS: updateList() updated https-etag to ' + newEtag)
-            } catch (e) {
-                console.log('HTTPS: updateList() error parsing server response')
-            }
-        })
     }
 
-    getFromStorage (fn) {
-        return new Promise((resolve, reject) => {
-            // For Chrome/Firefox:
-            if (window.chrome) {
-                console.log('HTTPS: getFromStorage() using chrome.storage.local (Chrome/FF)')
-
-                chrome.storage.local.get('https-upgrade-list', (results) => {
```

Comment:
I've been thinking about the possible upgrade path for users. I wonder whether we should delete any previous data that users on the current extension version might have.

Maybe we can add a check in `chrome-events`/`upgrade` reason: https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/b4856e22c809cf2dcad583f32cfa3438b5d835e8/shared/js/background/chrome-events.es6.js#L9-L16

... check if `https-upgrade-list` is in local storage, and delete it if yes.

Not sure if this is a big deal though.